src/main/java/com/example/java/epam/brayan/services/mappers/RefilledUserAccountMapper.java

```diff
@@ -0,0 +1,19 @@
+package com.example.java.epam.brayan.services.mappers;
+
+import com.example.java.epam.brayan.config.MappingConfig;
+import com.example.java.epam.brayan.data.entities.UserAccount;
+import com.example.java.epam.brayan.services.data.RefilledUserAccount;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+import org.springframework.stereotype.Component;
+
+@Mapper(config = MappingConfig.class)
+@Component
+public interface RefilledUserAccountMapper {
+    UserAccount refillUserAccount(RefilledUserAccount refilledUserAccount, @MappingTarget UserAccount userAccount);
+
+    default UserAccount addBalances(UserAccount userAccount, RefilledUserAccount refilledUserAccount) {
```

Comment:
I would not recommend to bloat the mappers with not related logic. Keep it simple. Just mapping operations. 

Move out the `addBalances` into the own service.